app/scripts/lib/createDupeReqFilterMiddleware.ts

```diff
@@ -52,14 +52,14 @@ export default function createDupeReqFilterMiddleware(): JsonRpcMiddleware<
   void
 > {
   const seenRequestIds = makeExpirySet();
-  return function filterDuplicateRequestMiddleware(req, _res, next, end) {
+  return function filterDuplicateRequestMiddleware(req, _res, next) {
     if (req.id === undefined) {
       // JSON-RPC notifications have no ids; our only recourse is to let them through.
-      return next();
-    } else if (!seenRequestIds.add(req.id)) {
+      next();
+    } else if (seenRequestIds.add(req.id)) {
+      next();
+    } else {
       log.info(`RPC request with id "${req.id}" already seen.`);
-      return end();
     }
```

Comment:
Per [my review](https://github.com/MetaMask/metamask-extension/pull/24672#pullrequestreview-2072557424), here's what I think will work:
```suggestion
    res.id = null;
    return end(new Error(`Received request with duplicate id "${req.id}".`));
    }
```